pages/gmail/return_email_url.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import print_function
+import pickle
+import os.path
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+import re
+import base64
+import email
+import urllib
+import urllib.request
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://www.googleapis.com/auth/gmail.readonly']
+
+
+def main():
+    creds = None
+    # The file token.pickle stores the user's access and refresh tokens, and is created automatically when
+    # the authorization flow completes for the first time.
+
+    if os.path.exists('token.pickle'):
+        with open('token.pickle', 'rb') as token:
+            creds = pickle.load(token)
+
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                'credentials.json', SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open('token.pickle', 'wb') as token:
+            pickle.dump(creds, token)
+
+    service = build('gmail', 'v1', credentials=creds)
+
+    # get inbox messages
+    results = service.users().messages().list(userId='me', labelIds=['INBOX']).execute()
+    messages = results.get('messages', [])
+
+    if not messages:
+        print('No tokens found.')
+    else:
+        # get list of URLs from last received email
+        for message in messages[:1]:
+
+            message = service.users().messages().get(userId='me', id=message['id'], format='raw').execute()
```

Comment:
Don't use loop iteration variable (message) in further processing. Here is not generating any damage, just a recommendation.

Reply:
Will keep in mind in the future